### DISCUSSION TITLE
Wizard not working after livewire v3 update

### DISCUSSION THREAD

**Discussion by rabol** [spatie/laravel-livewire-wizard#96] [Q&A]
Hi

I have 4 wizards in my project, 3 are working but the last is not

all 4 is registered in a serviceprovider

the one with problem like this:
```

        // Registration wizard
        Livewire::component('registration-wizard', RegistrationWizard::class);
        Livewire::component('registration-wizard-greeting-step', RegistrationWizardGreetingStep::class);
        Livewire::component('registration-wizard-information-step', RegistrationWizardInformationStep::class);
        Livewire::component('registration-wizard-verification-step', RegistrationWizardVerificationStep::class);
        Livewire::component('registration-wizard-summary-step', RegistrationWizardSummaryStep::class);
```

for some reason, the 'submit' method does not find the next step

```
    public function submit(): void
    {

        if ($this->hasNextStep()) {
            $this->nextStep();
        }

        dd('her');
    }
```

there is no error in the console, there is no error in the laravel log

**Comment by rabol** [comment]
I found it...

#[On('previousStep')]
and 
 #[On('nextStep')]

was missing as I override these two methods to the exception

**Reply by redafillali** [reply]
where did you add this?

**Reply by rabol** [reply]
in the WizardComponent

```
#[On('nextStep')]
public function nextStep(array $currentStepState): void
{
...
}
```

**Reply by redafillali** [reply]
worked perfectly, thank you so much

**Reply by redafillali** [reply]
> in the WizardComponent
> 
> ```
> #[On('nextStep')]
> public function nextStep(array $currentStepState): void
> {
> ...
> }
> ```
I did this

```
  #[On('nextStep')]
  public function nextStep($step)
  {
    // get current step
    // Go to the next step
    $this->showStep($step);
  }
```
but when I wna't to get state, I return emty array

**Reply by rabol** [reply]
Difficult to see the issue without the complete code and what you expect to see
